### PR TITLE
BinaryMarshaler / BinaryUnmarshaler

### PR DESCRIPTION
**Pull request by balintb**

### CODE REVIEW

**Comment by nieksand** [comment]
Needed (de)serialization support in a project I'm doing so I played with this branch.

Three notes:

1. It would be nice to have unmarshal/marshal use Writer and Reader rather than force the []byte.  If you really want an in-memory byte representation, you would use http://golang.org/pkg/bytes/#Buffer
2. Keeping hll++ representation as-is would be keen
3. Adding a version byte at the beginning and sanity checking against it might be a nice safety catch for when future versions of the lib break binary compatability.

File: hyperloglog.go [hunk 1]
```diff
@@ -81,3 +83,49 @@ func (h *HyperLogLog) Count() uint64 {
 	}
 	return uint64(-two32 * math.Log(1-est/two32))
 }
+
+// BinaryMarshaler marshals the HyperLogLog into binary form for storage
+func (h *HyperLogLog) BinaryMarshaler() ([]byte, error) {
```
Comment (clarkduvall) [inline hyperloglog.go]: I think this should be called MarshalBinary, and the other called UnmarshalBinary to implement the BinaryMarshaler and BinaryUnmarshaler interface

http://golang.org/pkg/encoding/

**Comment by nieksand** [comment]
I put together an alternative implementation based on my notes: https://github.com/clarkduvall/hyperloglog/pull/12

**Comment by clarkduvall** [comment]
@balintb would the solution in https://github.com/clarkduvall/hyperloglog/pull/12#issuecomment-87331850 work for you?

**Comment by balintb** [comment]
Absolutely, apologies, I'm a bit late to the party. @nieksand yeah good notes!

**Comment by clarkduvall** [comment]
closing as https://github.com/clarkduvall/hyperloglog/pull/15 adds this functionality

**Pull request closed without merging**